### PR TITLE
Add gitignore support for the Zig programming language

### PR DESCRIPTION
**Pull request by timmyjose**
**Reasons for making this change:**

We already have `linguist` support for Zig, but we do not have a `.gitignore` template for Zig projects created under Github. This PR aims to provide support for that.

**Links to documentation supporting these rule changes:**

Source for `.gitignore` template - https://github.com/ziglang/zig/blob/master/.gitignore

 **Link to application or project’s homepage**:
https://ziglang.org (homepage)
https://github.com/ziglang/zig/ (main repo page)

### CODE REVIEW

**Comment by timmyjose** [comment]
Hello, I'm just curious - is there anything blocking this PR from being merged, or is there any other reason for keeping it open?

**Comment by IridescentRose** [comment]
I'd love to see this happen

File: Zig.gitignore
Comment (ifreund) [inline Zig.gitignore:7]: None of these belong in a generic zig gitignore template, they are cmake/c++ related. Only `zig-cache` and `zig-out` should be included.

**Review by ifreund — CHANGES_REQUESTED** [review]

File: Zig.gitignore
Comment (snxx-lppxx) [reply Zig.gitignore:7]: Maybe also `*.pdb` and `*.exe`?

**Comment by snxx-lppxx** [comment]
Hi @bdougie, please review this.

File: Zig.gitignore
Comment (ifreund) [reply Zig.gitignore:7]: no

Comment (snxx-lppxx) [reply Zig.gitignore:7]: But Zig creates them when compiling...

Comment (ifreund) [reply Zig.gitignore:7]: Are you using the zig build system? I assume not as if you were they would be placed in `zig-out`.

There is no way to know what organization projects not using the zig build system will use, so they should IMO not be handled here.

Comment (snxx-lppxx) [reply Zig.gitignore:7]: The Zig build system is a project outside of the main compiler, and those who don't use this system should also be taken into account.

Comment (ifreund) [reply Zig.gitignore:7]: > The Zig build system is a project outside of the main compiler, 

No, it's part of the main compiler, see the `zig build` command.

> and those who don't use this system should also be taken into account.

There's no way to take them into account, as there is no standard place artifacts are stored if not using the zig build sysetm.

Comment (ifreund) [reply Zig.gitignore:7]: Anyhow, I'm not the target audience of this repo. I just add whatever files make sense to my `.gitignore` manually. As such, I don't really care what gets merged here personally. I am however a member of the zig core team.

**Comment by bdoner** [comment]
This should not be merged as is. The changes does not reflect what artifacts `zig build` creates.

The `zig build` command ([the built in build system that comes with Zig](https://ziglang.org/documentation/0.9.1/#Zig-Build-System)) only makes two directories those being `zig-out` containing the final libraries or binaries and the `zig-cache` which contains built artifacts helping to speed up subsequent builds.

The `build`, `build-*` and `docgen_tmp` directories in @timmyjose's PR are not standard directories when using Zig as the end 
user.

The Zig .gitignore file should really be nothing but:
```
# This file is for zig-specific build artifacts.

zig-cache/
zig-out/
```

**Comment by AliChraghi** [comment]
@timmyjose has not committed/commented for almost two years. i think we have to make another PR

**Comment by bdoner** [comment]
I've taken the liberty to open a new PR #3975 due to Timmy being inactive on Github.

**Comment by timmyjose** [comment]
> I've taken the liberty to open a new PR #3975 due to Timmy being inactive on this issue.

Agreed. I'm not active in the Zig community anymore. I will close this PR seeing that you've created a new PR for that. Thanks!

**Comment by timmyjose** [comment]
Closing in lieu of https://github.com/github/gitignore/pull/3975

**Pull request closed without merging**
